src/Essentials/src/FileSystem/FileSystem.uwp.cs

```diff
@@ -20,10 +39,7 @@ string PlatformCacheDirectory
 				? ApplicationData.Current.LocalCacheFolder.Path
 				: Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), AppSpecificPath, "Cache");
 
-		string PlatformAppDataDirectory
-			=> AppInfoUtils.IsPackagedApp
-				? ApplicationData.Current.LocalFolder.Path
-				: Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), AppSpecificPath, "Data");
+		string PlatformAppDataDirectory => _platformAppDataDirectory;
```

Comment:
Maybe instead of moving the logic the constructor, it might be better to do all the checks and creating in this property? So this property will throw vs the ctor.